src/test/recovery/t/121_streaming_and_archiving.pl

```diff
@@ -0,0 +1,222 @@
+use strict;
+use warnings;
+use PostgreSQL::Test::Cluster;
+use PostgreSQL::Test::Utils;
+use Test::More;
+
+my $node_primary;
+my $node_standby;
+
+##################### Test with wal_sender_archiving_status_interval disabled #####################
+
+# Initialize primary node with WAL archiving setup and wal_sender_archiving_status_interval as disabled
+$node_primary = PostgreSQL::Test::Cluster->new('primary');
+$node_primary->init(
+	has_archiving    => 1,
+	allows_streaming => 1);
+$node_primary->append_conf('postgresql.conf', 'wal_sender_archiving_status_interval = 0');
+$node_primary->start;
+my $primary_data = $node_primary->data_dir;
+
+# Set an incorrect archive_command so that archiving fails
+my $incorrect_command = "exit 1";
+$node_primary->safe_psql(
+'postgres', qq{
+ALTER SYSTEM SET archive_command TO '$incorrect_command';
+SELECT pg_reload_conf();
+});
+
+# Take backup
+$node_primary->backup('my_backup');
+
+# Initialize the standby node. It will inherit wal_sender_archiving_status_interval from the primary
+$node_standby = PostgreSQL::Test::Cluster->new('standby');
+$node_standby->init_from_backup($node_primary, 'my_backup', has_streaming => 1);
+my $standby_data = $node_standby->data_dir;
+$node_standby->start;
+
+$node_primary->safe_psql(
+	'postgres', q{
+	CREATE TABLE test1 AS SELECT generate_series(1,10) AS x;
+});
+
+my $current_walfile = $node_primary->safe_psql('postgres', "SELECT pg_walfile_name(pg_current_wal_lsn())");
+$node_primary->safe_psql(
+	'postgres', q{
+	SELECT pg_switch_wal();
+	CHECKPOINT;
+});
+
+# After switching wal, the current wal file will be marked as ready to be archived on the primary. But this wal file
+# won't get archived because of the incorrect archive archive_command
+my $walfile_ready = "pg_wal/archive_status/$current_walfile.ready";
+my $walfile_done = "pg_wal/archive_status/$current_walfile.done";
+
+# Wait for the standby to catch up
+$node_primary->wait_for_catchup($node_standby, 'replay',
+	$node_primary->lsn('insert'));
+
+# Wait for archive failure
+$node_primary->poll_query_until('postgres',
+q{SELECT failed_count > 0 FROM pg_stat_archiver}, 't')
+or die "Timed out while waiting for archiving to fail";
+
+# Due to archival failure, check if primary has only .ready wal file and not .done
+# Check if standby has .done created as wal_sender_archiving_status_interval is disabled.
+ok( -f "$primary_data/$walfile_ready",
+        ".ready file exists on primary for WAL segment $current_walfile"
+);
+ok( !-f "$primary_data/$walfile_done",
+        ".done file does not exist on primary for WAL segment $current_walfile"
+);
+
+ok( !-f "$standby_data/$walfile_ready",
+		".ready file does not exist on standby for WAL segment $current_walfile when wal_sender_archiving_status_interval=0"
```

Comment:
Need to remove a tab.
```suggestion
	".ready file does not exist on standby for WAL segment $current_walfile when wal_sender_archiving_status_interval=0"
```